presto-common/src/main/java/com/facebook/presto/common/zorder/ZOrderIntegers.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.common.zorder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.lang.String.format;
+
+public class ZOrderIntegers
+        implements ZOrder
+{
+    public static final int MAX_INPUT_SIZE = 10;
+
+    // Assume mostSignificantSetBitPositions is 1-based
+    private List<Integer> mostSignificantSetBitPositions;
+
+    public ZOrderIntegers(List<Integer> mostSignificantSetBitPositions)
+    {
+        this.mostSignificantSetBitPositions = mostSignificantSetBitPositions;
+    }
+
+    private void checkInputValidity(List<Integer> input)
+            throws IllegalArgumentException
+    {
+        if (input.size() == 0) {
+            throw new IllegalArgumentException("Input list size should be greater than zero.");
+        }
+
+        if (input.size() > MAX_INPUT_SIZE) {
+            throw new IllegalArgumentException("Our current Z-Ordering implementation does not support more than 10 input numbers.");
```

Comment:
10 should replaced with MAX_INPUT_SIZE. BTW, should we have a better name for "input"? I know I gave this name, but when I read the code, input sounds a really bad name.

Reply:
Could you please replace 10 with MAX_INPUT_SIZE?

Reply:
I mean in the error message the "10" should be "MAX_INPUT_DIMENSIONS".

Reply:
Oh, I didn't catch that! Thanks!